### PR TITLE
Basic config form

### PR DESCRIPTION
**Pull request by cquiroz**
This PR builds the first form on the configurations tile. This form should have an imaging alternate but at this level it only has spectroscopy

Note there is no persistence or any real use of this data but this PR will let us get feedback on the UI

image

### CODE REVIEW

File: model/shared/src/main/scala/explore/model/SpectroscopyConfigurationOptions.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import eu.timepit.refined.types.numeric.PosInt
+import eu.timepit.refined.cats._
+import explore.model.enum.FocalPlaneOptions
+import explore.model.enum.SpectroscopyCapabilities
+import lucuma.core.math.Angle
+import lucuma.core.math.Wavelength
+import monocle.macros.Lenses
+
+@Lenses
+final case class SpectroscopyConfigurationOptions(
```
Comment (rpiaggio) [inline model/shared/src/main/scala/explore/model/SpectroscopyConfigurationOptions.scala:17]: We will probably switch to using generated classes once there is a model to generate classes from.  But in the meanwhile we do need this.

**Review by rpiaggio — APPROVED** [review]
LGTM

File: model/shared/src/main/scala/explore/model/SpectroscopyConfigurationOptions.scala (hunk 1 above)
Comment (cquiroz) [reply model/shared/src/main/scala/explore/model/SpectroscopyConfigurationOptions.scala:17]: Indeed, and it wouldn't be stored in the state but rather come from a query

**Review by toddburnside — APPROVED** [review]
LGTM

**Pull request merged**